spyder/plugins/tests/test_history.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+import pytest
+
+from qtpy.QtGui import QTextOption
+
+from spyder.plugins import history
+
+
+#==============================================================================
+# Constants
+#==============================================================================
+
+
+#==============================================================================
+# Utillity Functions
+#==============================================================================
+options = {'wrap': False,
+           'line_numbers': False,
+           'go_to_eof': True,
+           'max_entries': 100}
+
+
+def get_option(self, option):
+    global options
+    return options[option]
+
+
+def set_option(self, option, value):
+    global options
+    options[option] = value
+
+
+#==============================================================================
+# Qt Test Fixtures
+#==============================================================================
+@pytest.fixture
+def historylog(qtbot):
+    historylog = history.HistoryLog(None, testing=True)
+    qtbot.addWidget(historylog)
+    historylog.show()
+    yield historylog
+    historylog.closing_plugin()
+    historylog.close()
+
+
+@pytest.fixture
+def historylog_with_tab(historylog, mocker, monkeypatch):
+    hl = historylog
+    # Mock read so file doesn't have to exist.
+    mocker.patch.object(history.encoding, 'read')
+    history.encoding.read.return_value = ('', '')
+
+    # Monkeypatch current options.
+    monkeypatch.setattr(history.HistoryLog, 'get_option', get_option)
+    monkeypatch.setattr(history.HistoryLog, 'set_option', set_option)
+
+    # Create tab for page.
+    hl.set_option('wrap', False)
+    hl.set_option('line_numbers', False)
+    hl.set_option('max_entries', 100)
+    hl.set_option('go_to_eof', True)
+    hl.add_history('test_history.py')
+    return hl
+
+
+#==============================================================================
+# Tests
+#==============================================================================
+
+def test_init(historylog):
```

Comment:
Please add a short docstring to each test to describe what each one is testing.

Reply:
Done.  My tests matched to the methods, so I wasn't quite sure what to write.  If it's too redundant, I can change it.